perfkitbenchmarker/flag_util.py

```diff
@@ -190,3 +191,149 @@ def DEFINE_integerlist(name, default, help, on_nonincreasing=None,
   serializer = IntegerListSerializer()
 
   flags.DEFINE(parser, name, default, help, flag_values, serializer, **args)
+
+
+SI_SUFFIXES = ['', 'K', 'M', 'G', 'T', 'P']
+
+
+@functools.total_ordering
+class ObjectSize(object):
+  """Holds the size of data, i.e. numbers of bytes.
+  """
+
+  def __init__(self, bytes=0):
+    self.bytes = bytes
+
+  def mult_for_base_and_exp(self, base, exp):
+    """Given base and exp, returns mult so that
+    bytes = mult * base ^ exp.
+
+    May return a float.
+    """
+
+    base_to_exp = base ** exp
+    return float(self.bytes) / base_to_exp
+
+  def exact_mult_for_base_and_exp(self, base, exp):
+    """Given base and exp, returns mult so that
+    bytes = mult * base ^ exp.
+
+    Raises ValueError if mult is not an integer.
+    """
+
+    base_to_exp = base ** exp
+    if self.bytes % base_to_exp == 0:
+      return self.bytes / base_to_exp
+    else:
+      raise ValueError('Bytes %s not divisible by %s',
+                       self.bytes, base_to_exp)
+
+  def ExactB(self):
+    return self.bytes
+
+  def ExactKiB(self):
+    return self.exact_mult_for_base_and_exp(1024, 1)
+
+  def ExactMiB(self):
+    return self.exact_mult_for_base_and_exp(1024, 2)
+
+  def ExactGiB(self):
+    return self.exact_mult_for_base_and_exp(1024, 3)
+
+  def AsB(self):
+    return self.bytes
+
+  def AsKiB(self):
+    return self.mult_for_base_and_exp(1024, 1)
+
+  def AsMiB(self):
+    return self.mult_for_base_and_exp(1024, 2)
+
+  def AsGiB(self):
+    return self.mult_for_base_and_exp(1024, 3)
+
+  def WriteWithBase(self, base):
+    """Write bytes in scientific notation.
+
+    Find mult and exp so that self.bytes = mult * base ^ exp,
+    where exp is as large as possible but <= len(SI_SUFFIXES)
+
+    Returns:
+      A tuple (mult, exp).
+    """
+
+    num = self.bytes
+    exp = 0
+
+    if num == 0:
+      return 0, 0
+
+    while num % base == 0 and exp < len(SI_SUFFIXES):
+      num = num // base
+      exp += 1
+
+    return num, exp
+
+  def __str__(self):
+    mult_ten, exp_ten = self.WriteWithBase(1000)
+    mult_two, exp_two = self.WriteWithBase(1024)
+
+    if exp_ten >= exp_two:
```

Comment:
Worth noting, a value like 1024000 will be "1024KB", rather than "1KiB". I think this can lead to confusion, and the preference should probably be reversed. Either way, I suggest a test case.